### ISSUE TITLE
Incorrect string formatting in distance display during navigation

### ISSUE THREAD

**Issue by lailaglez** [mapbox/mapbox-navigation-android#1793]
Route distance displayed as 4.5 %1$.1f mil instead of the expected 4.5 mil.

**Android API:** Tested on phone with API 26 and emulator with API 28. Both devices have en_US Locale.
**Mapbox Navigation SDK version:**
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.30.0'

### Steps to trigger behavior

 1. StartNavigation
        NavigationLauncherOptions options = NavigationLauncherOptions.builder()
                .directionsRoute(currentRoute)
                .build();
        NavigationLauncher.startNavigation(getActivity(), options);

![capture](https://user-images.githubusercontent.com/16417732/53904762-c5807600-4081-11e9-93f2-97db0079535a.PNG) ![wechat image_20190307022554](https://user-images.githubusercontent.com/16417732/53904643-87835200-4081-11e9-9d0d-96d8070a1e7f.jpg)

Both the emulator and the phone have en_US as their default Locale.

**Comment by danesfeder** [comment]
Hey @leylafenix 👋 thanks for the feedback!  Can you provide the device `Locale` and expected output so we can reproduce and fix?  Thanks!

**Comment by leylafenix** [comment]
Thanks for the quick reply @danesfeder.
Both the emulator and the phone have en_US as their default Locale. The distance is displayed as _4.5 %1$.1f mil_ instead of the expected _4.5 mil_.

**Comment by leylafenix** [comment]
So @danesfeder, I figured out what the issue was. Turns out I had a `miles` string defined in my strings resource file, and it seems that the name was being used by Mapbox navigation library. I simply renamed my string and that resolved the issue, but I do think this should not happen, so maybe the following info can help you guys look into it.  
This was my string definition:
```
<string name="miles">%1$.1f mi</string>
```
This was the notice I got when I tried to rename it using refactoring, with the library collisions:
``` 
The resource @string/miles is defined outside of the project (in one of the libraries) and cannot be updated. This can change the behavior of the application.  Are you sure you want to do this?  
Unhandled references: 

C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-bg\values-bg.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-cs\values-cs.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-de\values-de.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-es\values-es.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-fr\values-fr.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-hu\values-hu.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-ko\values-ko.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-my\values-my.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-pt-rBR\values-pt-rBR.xml 
C:\Users\*\.gradle\caches\transforms-1\files-1.1\jetified-mapbox-android-navigation-0.30.0.aar\0e6773449b5ebe0879cdfe4b0d7306e4\res\values-pt-rPT\values-pt-rPT.xml
... (Additional results truncated)
```
Hope this helps. Thanks.

**Comment by danesfeder** [comment]
@leylafenix yeah this helps - we are going to look into internalizing our resources that we use or adding a prefix that is more specific to the project to lessen the chances of something like this happening again.  Thanks the debugging and help looking into that!

**Comment by Guardiola31337** [comment]
@abhishek1508 Could we make sure this is addressed in `1.0`?

**Comment by Guardiola31337** [comment]
@abhishek1508 is this going to be addressed in the `1.0` UI SDK?

cc @JunDai @cafesilencio

**Comment by abhishek1508** [comment]
Let's fix.
cc @JunDai @cafesilencio @langsmith
